ibm/mas_devops/roles/dro/tasks/install/main.yml

```diff
@@ -0,0 +1,284 @@
+---
+
+# 1. Check for MAS ImageContentSourcePolicy
+# -----------------------------------------------------------------------------
+- name: "Determine whether this is an airgap environment"
+  include_tasks: "{{ role_path }}/../../common_tasks/detect_airgap.yml"
+
+
+# 2. Load default storage class (if not provided by the user)
+# -----------------------------------------------------------------------------
+- include_tasks: tasks/install/determine-storage-classes.yml
+
+- name: "DRO Deployment details"
+  debug:
+    msg:
+      - "DRO Namespace ...................... redhat-marketplace"
+      - "DRO Storage Class .................. {{ dro_storage_class }}"
+      - "DRO Pull Secret Token............... Can get from https://marketplace.redhat.com/en-us/account/keys"
+      - "DRO CSV Version..................... {{ dro_version }}"
+      - "MAS Instance ID .................... {{ mas_instance_id | default('<undefined>', True) }}"
+      - "MAS Config Directory ............... {{ mas_config_dir | default('<undefined>', True) }}"
+      - "DRO Contact First Name ............. {{ dro_contact.first_name | default('<undefined>', True) }}"
+      - "DRO Contact Last Name .............. {{ dro_contact.last_name | default('<undefined>', True) }}"
+      - "DRO Contact e-mail ................. {{ dro_contact.email | default('<undefined>', True) }}"
+
+
+# 3. Install DRO Prerequisites
+# -----------------------------------------------------------------------------
+
+# Check redhat-marketplace operator exists
+# -----------------------------------------------------------------------------
+- name: Check if operator group is present in redhat-marketplace already
+  kubernetes.core.k8s_info:
+    namespace: redhat-marketplace
+    kind: OperatorGroup
+  register: og_info
+
+- name: "Create OperatorGroup in redhat-marketplace namespace if not exist"
+  kubernetes.core.k8s:
+    definition: "{{ lookup('template', 'templates/operatorgroup.yml.j2') }}"
+
+# Create Marketplace Pull Secret
+- name: Get marketplace secret 
```

Comment:
This task is to check whether the secret is already created or not .If yes then we are not using `rm_sec` info.
I think we can use this info in task `Create Secret redhat-marketplace-pull-secret` and in `check pull secret` where we don't fail if secret has been already created .

Reply:
Updated conditions of mentioned tasks to use `rm_sec` check